### PR TITLE
Sep6 99 be add custom movie list functions

### PR DESCRIPTION
**Pull request by HolaSombreros**
Azure functions for add/delete movie list, add/delete movie from movie list, get all movie lists for a user

All tested in postman

## Add Movie List
image
image

## Get Movie Lists
image
image

## Delete Movie List
image
image

## Add Movie to Movie List
image
image

## Delete Movie From Movie List
image
image

### CODE REVIEW

File: src/MovieManagement.Database/Repositories/MovieListRepository.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public MovieListRepository(MovieManagementDbContext context, IRepository<MovieListEntity?> repository) {
+        _context = context;
+        _repository = repository;
+    }
+
+    public async Task<IList<MovieListEntity>> GetMovieListsByUserAsync(Guid userId) {
+        var list = await _context.MovieLists.Select(m => m).Where(m => m.UserId.Equals(userId)).ToListAsync();
+        return list;
+    }
+
```
Comment (vladdirlea2002) [inline src/MovieManagement.Database/Repositories/MovieListRepository.cs:16]: Three spaces no shame =))

File: src/MovieManagement.Functions/Dtos/AddMovieListDto.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace MovieManagement.Functions.Dtos; 
+
+public class AddMovieListDto {
```
Comment (vladdirlea2002) [inline src/MovieManagement.Functions/Dtos/AddMovieListDto.cs:3]: Do we really need this class? Could we not use the MovieListDto below?

File: src/MovieManagement.Functions/Dtos/MovieToMovieListDto.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace MovieManagement.Functions.Dtos;
+
+public class MovieToMovieListDto
```
Comment (vladdirlea2002) [inline src/MovieManagement.Functions/Dtos/MovieToMovieListDto.cs:3]: Same here (using movieListDto)

File: src/MovieManagement.Functions/MovieList/AddMovieToMovieList.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                MovieListId = movieListId,
+                MovieId = movie.MovieId
+            };
+
+            var result = await _validator.ValidateAsync(request);
+            if (!result.IsValid)
+            {
+                log.LogInformation("Body request not valid" + result.Errors[0].ErrorMessage);
+                return new BadRequestObjectResult(result.Errors);
```
Comment (vladdirlea2002) [inline src/MovieManagement.Functions/MovieList/AddMovieToMovieList.cs:52]: Are we gonna keep all these logs in the end?

**Review by vladdirlea2002 — APPROVED** [review]

File: src/MovieManagement.Functions/Dtos/AddMovieListDto.cs (hunk 2 above)
Comment (HolaSombreros) [reply src/MovieManagement.Functions/Dtos/AddMovieListDto.cs:3]: but they have different properties

File: src/MovieManagement.Functions/Dtos/MovieToMovieListDto.cs (hunk 3 above)
Comment (HolaSombreros) [reply src/MovieManagement.Functions/Dtos/MovieToMovieListDto.cs:3]: This is mainly to encapsulate and validate the fields, but between movielistDto and this one they have different properties so it doesn't make sense to use the same thing everywhere

File: src/MovieManagement.Functions/MovieList/AddMovieToMovieList.cs (hunk 4 above)
Comment (HolaSombreros) [reply src/MovieManagement.Functions/MovieList/AddMovieToMovieList.cs:52]: Probably yes. They are useful to see in the Azure portal when you want to check if something went wrong and cannot debug it

**Pull request merged**